core/src/test/java/org/apache/metamodel/query/FromItemTest.java

```diff
@@ -98,4 +105,35 @@ public void testSubQueryJoinToString() throws Exception {
 				"MetaModelSchema.project a LEFT JOIN (SELECT c.contributor_id, c.project_id AS foobar, c.name FROM MetaModelSchema.role c) b ON a.project_id = b.foobar",
 				from.toString());
 	}
+	
+    public void testCompoundJoin() {
+        final Schema schema = getExampleSchema();
+   
+        final QueryPostprocessDataContext dc = new QueryPostprocessDataContext() {
+            @Override
+            protected DataSet materializeMainSchemaTable(Table table, Column[] columns, int maxRows) {
+                Object[] values = new Object[columns.length];
```

Comment:
Implementing this method seems needless since the query is never executed. So why not just throw UnsupportedOperationException here?